### PR TITLE
Added some python scripts to run encode jobs and plot rate distortion

### PR DESCRIPTION
**Pull request by zhouwei0625**

### CODE REVIEW

File: scripts/encapp_tests.py [hunk 1]
```diff
@@ -0,0 +1,390 @@
...
+
+    if len(device_serials) == 0:
+        print ("No device connected")
+        exit(0)
+    elif len(device_serials) > 1:
+        if (serial_no is None) :
+            print ("More than one devices connected. Please specifiy a device serial number")
+            exit(0)
+        elif (serial_no not in device_serials):
+            print ('Specified serial number "{}" is invalid.'.format(serial_no) )
```
Comment (chemag) [inline scripts/encapp_tests.py]: you should be coherent with using either " or ' for strings. FB's recommendation IIRC is single quotes

File: scripts/encapp_tests.py [hunk 2]
```diff
@@ -0,0 +1,390 @@
...
+            exit(0)
+    else:
+        serial_no = serial
+
+    #get device model
```
Comment (chemag) [inline scripts/encapp_tests.py]: did you run `flake8` ? I'd expect this to be not ok

File: scripts/encapp_tests.py [hunk 3]
```diff
@@ -0,0 +1,390 @@
...
+    line = re.findall(serial_no+'.*$',stdout,re.MULTILINE)
+    model_re = re.compile(r'model:(?P<model>\S+)')
+    match = model_re.search(line[0])
+    device_model = match.group('model')
+
+    #remove any files that are generated in previous runs
+    adb_cmd = 'adb -s ' + serial_no + ' shell ls /sdcard/'
+    stdout,stderr = run_cmd(adb_cmd)
+    output_files = re.findall(ENCAPP_OUTPUT_FILE_NAME_RE,stdout,re.MULTILINE)
```
Comment (chemag) [inline scripts/encapp_tests.py]: ... _RE, std...

Add a space after a comma

File: scripts/encapp_tests.py [hunk 4]
```diff
@@ -0,0 +1,390 @@
...
+    try:
+        print(cmd, sep=' ')
+        process = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        stdout, stderr = process.communicate()
+        # print ('standard output: {}'.format(stdout.decode()))
+        # print ('standard error: {}'.format(stderr.decode()))
+    except:
+        raise Exception('Failed to run adb command')
+
+    return stdout.decode(), stderr.decode()
```
Comment (chemag) [inline scripts/encapp_tests.py]: return retcode too (so the caller can know whether anything failed)

Also, are you sure `process.communicate` wait3's ?

File: scripts/encapp_tests.py [hunk 5]
```diff
@@ -0,0 +1,390 @@
...
+    args = parser.parse_args()
+
+    if len(sys.argv) == 1:
+        parser.print_help()
+        sys.exit()
+
+    main(args)
+
+
+
```
Comment (chemag) [inline scripts/encapp_tests.py]: remove extra lines

File: scripts/encapp_tests.py (hunk 4 above)
Comment (zhouwei0625) [reply scripts/encapp_tests.py]: Process.communicate() waits for the process to terminate

**Pull request merged**